packages/react-data-grid/src/ColumnMetrics.ts

```diff
@@ -38,17 +40,17 @@ function setDefferedColumnWidths(columns: Column[], unallocatedWidth: number, mi
 
 function setColumnOffsets(columns: Column[]): void {
   let left = 0;
-  for (const column of columns) {
+  for (const column of columns as CalculatedColumn[]) {
```

Comment:
We can probably modify this method and return a list of calculated columns instead. wdyt?

Reply:
Yeah, maybe we can merge some together so we can do
```ts
function x(columns: Column[]): CalculatedColumn[] {
  const cc: CalculatedColumn[] = [];

  for (const column of columns) {
    const left = ...
    const width = ...
    cc.push({ ...column, left, width });
  }

  return cc;
}
```